website/src/documentation/programming-guide.md

```diff
@@ -1970,7 +1976,1078 @@ records.apply("WriteToText",
+schema. For example, it’s common to have ephemeral fields in a class that should not be included in a schema 
+(e.g. caching the hash value to prevent expensive recomputation of the hash), and `@SchemaIgnore` can be used to
+exclude these fields.
+
+In some cases it is not convenient to annotate the POJO class, for example if the POJO is in a different package that is
+not owned by the Beam pipeline author. In these cases the schema inference can be triggered programmatically in 
+pipeline’s main function as follows:
+
+```java
+ pipeline.getSchemaRegistry().registerPOJO(TransactionPOJO.class); 
+```
+
+##### **Java Beans**
+Java Beans are a de-facto standard for creating reusable property classes in Java. While the full 
+standard has many characteristics, the key ones are that all properties are accessed via getter and setter classes, and 
+the name format for these getters and setters is standardized. A Java Bean class can be annotated with 
+`@DefaultSchema(JavaBeanSchema.class)` and Beam will automatically infer a schema for this class. For example:
+
+```java
+@DefaultSchema(JavaBeanSchema.class)
+public class TransactionBean {
+  public TransactionBean() { … } 
+  public String getBank() { … }
+  public void setBank(String bank) { … }
+  public double getPurchaseAmount() { … }
+  public void setPurchaseAmount(double purchaseAmount) { … }
+}
+// Beam will automatically infer the correct schema for this PCollection. No coder is needed as a result.
+PCollection<TransactionBean> beans = readBeans();
+```
+
+The `@SchemaCreate` annotation can be used to specify a constructor or a static factory method, in which case the 
+setters and zero-argument constructor can be omitted.
+
+```java
+@DefaultSchema(JavaBeanSchema.class)
+public class TransactionBean {
+  @SchemaCreate
+  Public TransactionBean(String bank, double purchaseAmount) { … }
+  public String getBank() { … }
+  public double getPurchaseAmount() { … }
+}
+```
+
+`@SchemaFieldName` and `@SchemaIgnore` can be used to alter the schema inferred, just like with POJO classes.
+
+##### **AutoValue(())
+Java value classes are notoriously difficult to generate correctly. There is a lot of boilerplate you must create in 
+order to properly implement a value class. AutoValue is a popular framework for easily generating such classes by i
+mplementing a simple abstract base class.
+
+Beam can infer a schema from an AutoValue class. For example:
+
+```java
+@DefaultSchema(AutoValueSchema.class)
+@AutoValue
+public abstract class TransactionValue {
+  public abstract String getBank(); 
+  public abstract double getPurchaseAmount();
+}
+```
+
+This is all that’s needed to generate a simple AutoValue class, and the above `@DefaultSchema` annotation tells Beam to
+infer a schema from it. This also allows AutoValue elements to be used inside of `PCollection`s.
+
+`@SchemaFieldName` and `@SchemaIgnore` can be used to alter the schema inferred.
+
```

Comment:
Can we add Avro here too?

Reply:
yes - plan to add Avro and Proto in a followup PR